control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -0,0 +1,310 @@
+use crate::{etcd::EtcdStore, nodes::NodeList};
+use chrono::Utc;
+use common_lib::types::v0::{
+    store::{
+        switchover::{Operation, OperationState, SwitchOverSpec, SwitchOverTime},
+        SpecTransaction,
+    },
+    transport::{NexusId, VolumeId},
+};
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
+use utils::NVME_TARGET_NQN_PREFIX;
+
+use std::convert::TryFrom;
+
+/// Stage represents the steps for switchover request
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum Stage {
+    /// Initialize switchover request
+    Init,
+    /// Shutdown original/old nexus
+    ShutdownOrigin,
+    /// Create new nexus
+    ReconstructNexus,
```

Comment:
nit: 
ReconstructTarget,
SwitchOverTarget,
DeleteOldTarget

Reply:
let's try to use the word Target instead of Nexus since Nexus is more internal to the core agent.